sync-core/src/test/java/com/cloudant/sync/query/QueryFilterFieldsTest.java

```diff
@@ -0,0 +1,203 @@
+//  Copyright (c) 2015 Cloudant. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package com.cloudant.sync.query;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+
+import com.cloudant.sync.datastore.BasicDocumentRevision;
+import com.cloudant.sync.datastore.ConflictException;
+import com.cloudant.sync.datastore.DocumentBodyFactory;
+import com.cloudant.sync.datastore.DocumentRevision;
+import com.cloudant.sync.datastore.MutableDocumentRevision;
+import com.cloudant.sync.datastore.ProjectedDocumentRevision;
+import com.cloudant.sync.util.SQLDatabaseTestUtils;
+
+import org.junit.Assert;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class QueryFilterFieldsTest extends AbstractQueryTestSetUp {
+
+    @Override
+    public void setUp() throws SQLException {
+        super.setUp();
+        im = new IndexManager(ds);
+        assertThat(im, is(notNullValue()));
+        db = im.getDatabase();
+        assertThat(db, is(notNullValue()));
+        assertThat(im.getQueue(), is(notNullValue()));
+        String[] metadataTableList = new String[] { IndexManager.INDEX_METADATA_TABLE_NAME };
+        SQLDatabaseTestUtils.assertTablesExist(db, metadataTableList);
+
+        setUpBasicQueryData();
+    }
+
+    // When filtering fields on find
+
+    @Test
+    public void returnsFieldSpecifiedOnly() {
+        // query - { "name" : "mike" }
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        QueryResult queryResult = im.find(query, 0, Long.MAX_VALUE, Arrays.asList("name"), null);
+        for (DocumentRevision rev : queryResult) {
+            Map<String, Object> revBody = rev.getBody().asMap();
+            assertThat(revBody.keySet(), contains("name"));
```

Comment:
Yes the `contains` Hamcrest matcher method works just like `containsInAnyOrder` and actually means "contains and is limited to...in content".